src/components/setting-list.tsx

```diff
@@ -0,0 +1,83 @@
+"use client";
+
+import { ReactNode, useState } from "react";
+import Link from "next/link";
+
+type Props = {
+  path?: string;
+  text: string;
+  icon: ReactNode;
+  external?: boolean;
+};
```

Comment:
```suggestion
type Props = {
  title: string;
  icon: LucideIcon;
  onClick?: () => void;
};
```

text よりも titleのほうが良さそうです！
iconはlucide icon使用するようお願いします！

pathやexternalを引数で渡すよりも`onClick`を引数で渡して、このコンポーネントを使用する側でクリック時の処理を書くほうが拡張性を持たせる事ができ、このコンポーネントもスッキリするかと！